### PR TITLE
feat: Add onboarding screen

### PR DESCRIPTION
**Pull request by nabilbndev**
Created ```OnboardingScreen()``` and all of its basics widgets. Replaced ```CurrentForecastScreen()``` with ```OnboardingScreen()``` in ```main.dart```.
Now, the ```CurrentForecastScreen()``` is pushed on top of ```OnboardingScreen()``` when ```GetStartedButton()``` is tapped from ```OnboardingScreen()```.

### CODE REVIEW

File: lib/features/onboarding/onboarding_screen.dart [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+              ),
+              RichText(
+                textAlign: TextAlign.center,
+                text: TextSpan(
+                    style: TextStyle(
+                      fontWeight: FontWeight.w800,
+                      fontSize: 25,
+                      color: darkMode
+                          ? WeatherTheme.lightTheme.scaffoldBackgroundColor
+                          : WeatherTheme.darkTheme.scaffoldBackgroundColor,
```
Comment (Xazin) [inline lib/features/onboarding/onboarding_screen.dart:37]:
```suggestion
                      color: Theme.of(context).colorScheme.onBackground,
```

File: lib/features/onboarding/onboarding_screen.dart [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+              ),
+              const SizedBox(
+                height: 35,
+              ),
+              GetStartedButton(
+                onTap: () {
+                  Navigator.push(context, MaterialPageRoute(
+                    builder: (context) {
+                      return const CurrentForecastScreen();
+                    },
```
Comment (Xazin) [inline lib/features/onboarding/onboarding_screen.dart:74]:
```suggestion
                    builder: (_) => const CurrentForecastScreen(),
```

File: lib/features/onboarding/onboarding_screen.dart [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                      TextSpan(
+                        text: 'conditions and radars',
+                      )
+                    ]),
+              ),
+              const SizedBox(
+                height: 35,
+              ),
+              GetStartedButton(
+                onTap: () {
```
Comment (Xazin) [inline lib/features/onboarding/onboarding_screen.dart:70]: There are no other arguments to this button, and it's a very specific button. Should the onTap just be hardcoded like everything else in the button?

File: lib/features/onboarding/widgets/get_started_btn.dart [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:weather_app/app_state/dark_mode_provider.dart';
+import 'package:weather_app/theme/theme.dart';
+
+class GetStartedButton extends ConsumerWidget {
+  const GetStartedButton({required this.onTap, super.key});
+
+  final VoidCallback onTap;
+  @override
```
Comment (Xazin) [inline lib/features/onboarding/widgets/get_started_btn.dart:10]:
```suggestion
  final VoidCallback onTap;

  @override
```

File: lib/features/onboarding/onboarding_screen.dart [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+import 'package:weather_app/features/current_forecast/current_forecast_screen.dart';
+import 'package:weather_app/features/onboarding/widgets/get_started_btn.dart';
+import 'package:weather_app/theme/theme.dart';
+
+class OnboardingScreen extends ConsumerWidget {
+  const OnboardingScreen({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final darkMode = ref.watch(darkModeProvider);
```
Comment (Xazin) [inline lib/features/onboarding/onboarding_screen.dart:13]: This `darkModeProvider` is quite redundant. We should only use it to decide on what Theme to use, and then use `Theme.of(context)` throughout the app.

File: lib/features/onboarding/widgets/get_started_btn.dart [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  Widget build(BuildContext context, WidgetRef ref) {
+    final darkMode = ref.watch(darkModeProvider);
```
Comment (Xazin) [inline lib/features/onboarding/widgets/get_started_btn.dart:12]: Redundant

File: lib/features/onboarding/widgets/get_started_btn.dart [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+                : WeatherTheme.darkTheme.scaffoldBackgroundColor),
+        child: Center(
+          child: Text(
+            'Get Started',
+            style: TextStyle(
+                fontSize: 16,
+                fontWeight: FontWeight.w400,
+                color: darkMode
+                    ? WeatherTheme.darkTheme.scaffoldBackgroundColor
+                    : WeatherTheme.lightTheme.scaffoldBackgroundColor),
```
Comment (Xazin) [inline lib/features/onboarding/widgets/get_started_btn.dart:32]:
```suggestion
            color:  Theme.of(context).colorScheme.onBackground,
        child: Center(
          child: Text(
            'Get Started',
            style: TextStyle(
                fontSize: 16,
                fontWeight: FontWeight.w400,
                color: Theme.of(context).colorScheme.onBackground,
```

**Review by Xazin — COMMENTED** [review]
In `theme.dart`, add `final Color textColor`.

eg.
```
class ThemeColors {
  factory ThemeColors.light() => const ThemeColors(
        scaffoldBackgroundColor: Color(0xFFD8D8D8),
        textColor: Color(0xFF200D46),
      );

  factory ThemeColors.dark() => const ThemeColors(
        scaffoldBackgroundColor: Color(0xFF200D46),
        textColor: Color(0xFFD8D8D8),
      );

  const ThemeColors({
    required this.scaffoldBackgroundColor,
    required this.textColor,
  });

  final Color scaffoldBackgroundColor;
  final Color textColor;
}
```
Then in `WeatherTheme` class, do:
```
  static ThemeData get lightTheme {
    final themeColors = ThemeColors.light();

    return ThemeData.light().copyWith(
      scaffoldBackgroundColor: themeColors.scaffoldBackgroundColor,
      colorScheme: ThemeData.light().colorScheme.copyWith(
        onBackground: themeColors.textColor,
      ),
    );
  }
  ```
  Do the same for the dark mode.
  
  Now you can use `Theme.of(context).colorScheme.onBackground` instead of relying on `darkMode`.

File: lib/features/onboarding/widgets/get_started_btn.dart (hunk 7 above)
Comment (nabilbndev) [reply lib/features/onboarding/widgets/get_started_btn.dart:32]: If you apply ```color: Theme.of(context).colorScheme.onBackground``` for both ```BoxDecoration()``` and ```TextStyle()```,
text becomes unreadable as both container and the text within it becomes of same color.

For now IMHO changing TextStyle's color property to ```color: Theme.of(context).colorScheme.onPrimary``` gives the text somewhat readability. We can change it later.

I suppose, to address this issue, we have to create another property of type ```Color``` within ```ThemeColors``` class in ```theme.dart```. What do you think?

File: lib/features/onboarding/onboarding_screen.dart (hunk 3 above)
Comment (nabilbndev) [reply lib/features/onboarding/onboarding_screen.dart:70]: Do you mean we write the button in the following way?
```
class GetStartedButton extends ConsumerWidget {
  const GetStartedButton({super.key});

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    return GestureDetector(
      onTap: () {
        Navigator.push(
            context,
            MaterialPageRoute(
              builder: (_) => const CurrentForecastScreen(),
            ));
      },
      child: Container(
        height: 54,
        width: 263,
        decoration: BoxDecoration(
            borderRadius: const BorderRadius.all(Radius.circular(7)),
            color: Theme.of(context).colorScheme.onBackground),
        child: Center(
          child: Text(
            'Get Started',
            style: TextStyle(
                fontSize: 16,
                fontWeight: FontWeight.w400,
                color: Theme.of(context).colorScheme.onBackground),
          ),
        ),
      ),
    );
  }
}
```

File: lib/features/onboarding/widgets/get_started_btn.dart (hunk 7 above)
Comment (Xazin) [reply lib/features/onboarding/widgets/get_started_btn.dart]: Yeah

**Pull request merged**